auth/src/main/java/org/aerogear/mobile/auth/authenticator/oidc/OIDCAuthenticatorImpl.java

```diff
@@ -169,15 +171,22 @@ private void performLogout(final URL logoutUrl) {
         HttpRequest request = serviceModule.newRequest();
         request.get(logoutUrl.toString());
 
-        // Creates and handles the response
-        new AppExecutors().networkThread().execute(() -> request.execute());
-        //it doesn't matter if the logout request is successful or not, we should always delete the local tokens
-        //the remote session should be timed out eventually
-        authStateManager.save(null);
+        final HttpResponse httpResponse = request.execute();
+
+        httpResponse.onComplete(() -> {
+            if (httpResponse.getStatus() == HTTP_OK || httpResponse.getStatus() == HTTP_MOVED_TEMP) {
+                // delete the local tokens when the session with the OIDC has been terminated
+                authStateManager.save(null);
+                logoutCallback.onSuccess();
+            } else {
+                MobileCore.getLogger().error("Error Performing a Logout on the Remote OIDC Server: ", httpResponse.getRequestError());
```

Comment:
should we fail the logout action even if we can't log out the current session? I thought the session will be expired eventually?

Reply:
I mentioned briefly about this in https://github.com/aerogear/aerogear-android-sdk/issues/96, but I think it is better to give the end developer more control or how they handle logouts when they can't contact the server. If we always the delete the local tokens when calling logout(), it will be impossible for a user to do a true logout against the server when they come online, as we have deleted the identity token. Simply deleting the tokens locally will still mean that all the tokens and the keycloak session are still active, we just don't have the tokens anymore (that doesn't stop someone else from using them though). With SSO in gerneral, the expiry times could be really long and once we delete the tokens locally, they are gone - but the session/tokens are still valid unless we notify the server.

I was planning on exposing a `deleteTokens()` function so the developer can choose how they handle online/offline logouts. All tokens/sessions will expire eventually, whether they click a logout button or not, but the use of callbacks plus exposing some token deletion methods can help them reduce the timelines to revoke sessions/tokens. 

On a minor side not, it *may* help reduce the number of concurrent sessions for a single user and have less of an impact on things like active session metrics.

Let me know which approach you think is best.

Reply:
Yeah, I think you are right. Let the developers decide what they want to do.